Formula/libff.rb

```diff
@@ -0,0 +1,44 @@
+class Libff < Formula
+  desc "C++ library for Finite Fields and Elliptic Curves"
+  homepage "https://github.com/scipr-lab/libff"
+  url "https://github.com/scipr-lab/libff/archive/refs/tags/v0.2.1.tar.gz"
+  sha256 "630010841f6992ad6e4758bb079d248eab46cfe66a4c0391c11075f3e5387b60"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "openssl@1.1" => :build
+
+  depends_on "gmp"
+
+  def install
+    # bn128 is somewhat faster, but requires an x86_64 CPU
+    curve = Hardware::CPU.intel? ? "BN128" : "ALT_BN128"
+
+    # build libff dynamically. The project only builds statically by default
+    inreplace "libff/CMakeLists.txt", "STATIC", "SHARED"
```

Comment:
Not from what I can tell. The CMake docs say `...If present and true, this will cause all libraries to be built shared unless the library was explicitly added as a static library` but it is explicitely marked `STATIC` here: https://github.com/scipr-lab/libff/blob/v0.2.1/libff/CMakeLists.txt#L23-L25

Reply:
And I take it upstream isn't interested in supporting building `libff` as a shared library?

Reply:
I haven't reached out to them to ask.

Reply:
Looking at the upstream build, I think this patch isn't sound anyways. Here's the lines it replaces:

https://github.com/scipr-lab/libff/blob/c47bd617c4361aaec377eb38e53d47758ca2b9ad/CMakeLists.txt#L100-L106

...so it looks like it's really changing the `gmp` dependency from shared to static? The actual `libff` build itself is already marked as static, here:

https://github.com/scipr-lab/libff/blob/c47bd617c4361aaec377eb38e53d47758ca2b9ad/libff/CMakeLists.txt#L25-L70

@elopez has anyone confirmed recently that Echidna stills needs this patch? I suspect it'll build fine without it.

Reply:
No, it's patching this line:

https://github.com/scipr-lab/libff/blob/c47bd617c4361aaec377eb38e53d47758ca2b9ad/libff/CMakeLists.txt#L27

We want `libff/CMakeLists.txt`, and not the top-level `CMakeLists.txt`.

Reply:
Oh, I completely misread this -- I thought it was patching a shared library build to make it static, not the other way around.

Reply:
Would be good to follow this up upstream, especially since other package managers seem to be doing something similar here...